src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -278,6 +280,315 @@ struct mac_addr_list
   }
 };
 
+struct aim_counter
+{
+  using result_type = query::aim_counter::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbgIpData = boost::any_cast<query::aim_counter::debug_ip_data_type>(arg1);
+    auto pdev = get_pcidev(device);
+
+    // read counter values
+    std::string aimName("aximm_mon_");
+    aimName = aimName + std::to_string(dbgIpData->m_base_address);
+
+    std::string namePath = pdev->get_sysfs_path(aimName.c_str(), "name");
+
+    std::size_t pos = namePath.find_last_of('/');
+    std::string path = namePath.substr(0, pos+1);
+    path += "counters";
+
+    result_type retvalBuf(9, 0);
+
+    std::ifstream ifs(path.c_str());
+    if(!ifs) {
+      std::cout << "\nINFO: Incomplete AIM counter data in " << path << std::endl;
+      return retvalBuf;
+    }
+
+    const size_t sz = 256;
+    char buffer[sz];
+    std::memset(buffer, 0, sz);
+    ifs.getline(buffer, sz);
+
+    result_type valBuf;
+
+    while(!ifs.eof()) {
+      valBuf.push_back(strtoull((const char*)(&buffer), NULL, 10));
+      std::memset(buffer, 0, sz);
+      ifs.getline(buffer, sz);
+    }
+
+    ifs.close();
+
+    if(valBuf.size() < 13) {
+      std::cout << "\nINFO: Incomplete AIM counter data in " << path << std::endl;
+      while(valBuf.size() < 13) {
+        valBuf.push_back(0);
+      }
+    }
+
+    retvalBuf[0] = valBuf[0];
+    retvalBuf[1] = valBuf[1];
+    retvalBuf[2] = valBuf[4];
+    retvalBuf[3] = valBuf[5];
+    retvalBuf[4] = valBuf[8];
+    retvalBuf[5] = valBuf[9];
+    retvalBuf[6] = valBuf[10];
+    retvalBuf[7] = valBuf[11];
+    retvalBuf[8] = valBuf[12];
```

Comment:
Argh,  took me a while to notice the jump from 5->8; I was suggesting std::copy, which you could still do and split into two.  Either that or at least a comment.